src/modules/0.0.15/aws_vpc/index.ts

```diff
@@ -364,9 +364,13 @@ export const AwsVpcModule: Module2 = new Module2({
       const out = new NatGateway();
       out.connectivityType = nat.ConnectivityType as ConnectivityType;
       const natPublicAddress = nat.NatGatewayAddresses?.filter(n => !!n.AllocationId).pop();
-      if (natPublicAddress) {
-        out.elasticIp = await AwsVpcModule.mappers.elasticIp.db.read(ctx, natPublicAddress.AllocationId) ??
-          await AwsVpcModule.mappers.elasticIp.cloud.read(ctx, natPublicAddress.AllocationId);
+      if (natPublicAddress?.AllocationId) {
+        try {
+          out.elasticIp = await AwsVpcModule.mappers.elasticIp.db.read(ctx, natPublicAddress.AllocationId) ??
+            await AwsVpcModule.mappers.elasticIp.cloud.read(ctx, natPublicAddress.AllocationId);
+        } catch (error: any) {
+          if (error.Code === 'InvalidAllocationID.NotFound') return undefined;
```

Comment:
This just makes the Nat Gateway go away if the IP is invalid. Is that what we want?

Reply:
Yes, right? That is the misconfiguration, any other case could be a transient error or another issue and we might fail.

Reply:
why is that code repeated twice? don't get it...

Reply:
But if you simultaneously create a new elastic IP and a nat gateway in new records in the database, wouldn't it reasonably return this error that we should retry the nat gateway construction on in that case?

Reply:
Hmmm no, it should not be a problem, because if both are created at the same time, the NAT Gateway will fail if the ID is not there yet (committing that change explicitly after this comment). Here, mapping from the cloud, we make sure that if the cloudRecord has an `allocationId`, that id has to exist, otherwise is a misconfiguration.